R/loadFromMetaphlan.R

```diff
@@ -0,0 +1,188 @@
+#' Import Metaphlan results to \code{TreeSummarizedExperiment}
+#
+#' @param metaphlan a single \code{character} value defining the file
+#'   path of the Metaphlan file. The file must be in merged Metaphlan format.
+#'
+#' @param sample_meta a single \code{character} value defining the file
+#'   path of the sample metadata file. The file must be in \code{tsv} format
+#'   (default: \code{sample_meta = NULL}).
+#'   
+#' @param phy_tree a single \code{character} value defining the file
+#'   path of the phylogenetic tree.
+#'   (default: \code{phy_tree = NULL}).
+#'   
+#' @param ... additional arguments:
+#' \itemize{
+#'   \item{\code{removeTaxaPrefixes}:} {\code{TRUE} or \code{FALSE}: Should
+#'     taxonomic prefixes be removed? (default:
+#'     \code{removeTaxaPrefixes = FALSE})}
+#' }
+#'
+#' @details
+#' Import Metaphlan results. Input must be in merged Metaphlan format.
+#' Data is imported so that data at the lowest rank is imported as a 
+#' \code{TreeSE} object. Data at higher rank is imported as a
+#' \code{SE} objects which are stored to \code{altExp} of
+#' \code{TreeSE} object.
+#'
+#' @return  A
+#' \code{\link[TreeSummarizedExperiment:TreeSummarizedExperiment-class]{TreeSummarizedExperiment}}
+#' object
+#'
+#' @name loadFromMetaphlan
+#' @seealso
+#' \code{\link[=makeTreeSummarizedExperimentFromPhyloseq]{makeTreeSummarizedExperimentFromPhyloseq}}
+#' \code{\link[=makeSummarizedExperimentFromBiom]{makeSummarizedExperimentFromBiom}}
+#' \code{\link[=makeTreeSummarizedExperimentFromDADA2]{makeTreeSummarizedExperimentFromDADA2}}
+#' \code{\link[=loadFromQIIME2]{loadFromQIIME2}}
+#' \code{\link[=loadFromMothur]{loadFromMothur}}
+#'
+#' @export
+#' @author Leo Lahti and Tuomas Borman. Contact: \url{microbiome.github.io}
+#' 
+#' @references
+#'
+#' @examples
+#' \dontrun{
+#' # File path
+#' file_path <- "/data/metaphlan_result.txt"
+#' # Import data
+#' tse <- loadFromMetaphlan(file_path)
+#' # Data at the lowest rank
+#' tse
+#' # Data at higher rank is stored in altExp
+#' altExps(tse)
+#' # Higher rank data is in SE format, for example, Phylum rank
+#' altExp(tse, "Phylum")
+#' }
+#' 
+
+loadFromMetaphlan <- function(metaphlan, sample_meta = NULL, phy_tree = NULL, ...){
+    ################################ Input check ################################
+    if(!.is_non_empty_string(metaphlan)){
+        stop("'metaphlan' must be a single character value.",
+             call. = FALSE)
+    }
```

Comment:
Lets call the argument file, because thats what is, isn't it?

Reply:
metaphlan --> file